docs: l1->l2 message fees (#52)
starknet-community-libs/starknet-docs: docs/L1-L2 Communication/messaging-mechanism.md

## L1 → L2 Message Fees

An L1 → L2 message induces a transaction on L2 which is not associated with an account (unlike regular transactions). This calls for a different mechanism for paying the transaction's fee, for otherwise the sequencer has no incentive of including l1 handler calls inside a block.

> To avoid having to interact with both L1, L2 when sending a message, L1 → L2 messages are paypable on L1, by sending ETH with the call to the payable function `sendMessageToL2` on the StarkNet Core contract. This payment will be taken as a fee for handling the message, and will be charged by the sequencer in full upon updating the L1 state with the consumption of this message.

Reviewer: ```suggestion
To avoid having to interact with both L1 and L2 when sending a message, L1 → L2 messages are payable on L1, by sending ETH with the call to the payable function `sendMessageToL2` on the StarkNet Core contract. The sequencer takes this fee in exchange for handling the message. The sequencer charges the fee in full upon updating the L1 state with the consumption of this message.
```

> Note that throughtout the lifecycle of the message, the assoicated fee is locked in the Core contract. If the message was not processed for any reason, then [cancelling](./messaging-mechanism#l1--l2-message-cancellation) the message will

Reviewer: ```suggestion
:::Note
Throughout the lifecycle of the message, the associated fee is locked in the Core contract. If the message is not processed for any reason, then [cancelling](./messaging-mechanism#l1--l2-message-cancellation) the message
```

> also return the fee to the sender.

Reviewer: ```suggestion
also returns the fee to the sender.
```

> You can use the [CLI](../CLI/commands#starknet-estimate_message_fee) to get an estimation for a L1 → L2 message fee.

Reviewer: ```suggestion
You can use the [starknet estimate_message_fee](../CLI/commands#starknet-estimate_message_fee) command to get an estimate of a L1 → L2 message fee.
```
